server/sql_app/models.py

```diff
@@ -0,0 +1,67 @@
+""" Create SQLAlchemy models from the 'Base' class."""
+from sqlalchemy import DateTime, Column, ForeignKey, Integer, String
+from sqlalchemy.orm import relationship
+from sqlalchemy.sql import func
+
+from .database import Base
+
+# SQLAlchemy uses 'model' to refer to these classes and instances that
+# interact with the database
+# Pydantic uses term 'model' to refer to data validation, conversion, and
+# documentation classes and instances
+
+
+class Job(Base):
+    """SQLAlchemy model."""
+    __tablename__ = "jobs"
+
+    id = Column(Integer, primary_key=True, index=True)
+    name = Column(String, index=True)
+    description = Column(String, index=True)
+    status = Column(String, index=True)
+
+
+class User(Base):
+    """SQLAlchemy model."""
+    __tablename__ = "users"
+
+    id = Column(Integer, primary_key=True, index=True)
+    session_id = Column(String, unique=True, index=True)
+    last_active = Column(DateTime(timezone=True), server_default=func.now())
+
+    # 'relationship' a "magic" attribute that will contain the values from
+    # other tables related to this one.
+    # When accessing the attribute 'items' in a User, as in 'my_user.items',
+    # it will have a list of Item SQLAlchemy models (from the 'items' table)
+    # that have a foreign key pointing to this record in the 'users' table.
+
+    # When you access 'my_user.items', SQLA will actually go and fetch the items
+    # from the db in the 'items' table and populate them here.
+    scenarios = relationship("Scenario", back_populates="owner")
+    patterns = relationship("Pattern", back_populates="owner")
```

Comment:
Great question, my understanding is that it allows us to call on `my_user.scenarios` such that SQLAlchemy fetches the `scenarios` from the database in the `scenarios` table, and populates them there. Likewise, when calling `my_scenario.owner`, it will contain a `User` model from the `users` table. I actually think I could re-write some endpoints to take advantage of this. Maybe that can be done in a future cleanup round.

Better said here in the docs: https://fastapi.tiangolo.com/tutorial/sql-databases/#create-the-relationships

Reply:
This `relationship` thing in SQLAlchemy looks like a really interesting way to represent the relationships between tables, like a `JOIN` without us having to construct the SQL syntax. I'm still not 100% clear on what the `populates` part of this means, but I think I get what it's supposed to represent.